### DISCUSSION TITLE
❓Where would `autodiscoverFilter` configuration live?

### DISCUSSION THREAD

**Discussion by greg-slonim** [renovatebot/renovate#23580] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.24.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

## Question
I would like to find out where I can setup the repo filter using `autodiscoverFilter`

## Context
I have a hosted environment with
renovate-bot
  |
  |- renovate-config (with shared configuration)
  |- renovate-runner

I currently pass the `autodiscoverFilter` value either via `RENOVATE_EXTRA_FLAGS` (with `--autodiscoverFilter`) or `RENOVATE_AUTODISCOVER_FILTER` CI/CD env variable.

It is a bit painful to maintain. I attempted to use `autodiscoverFilter` in the actual runner's config (and removed the env variables) but afterwards Renovate started discovering all repos its token has access to.

## Question
Where [as in, which configuration] I can use to host the `autodiscoverFilter` array?

### What you tried so far.

Include `autodiscoverFilter` in renovate-runner configuration

### Relevant debug logs

<details><summary>Logs</summary>

This log was produced using 
```
RENOVATE_EXTRA_FLAGS: --autodiscover=true --onboarding=true --autodiscover-filter=thetradedesk/teams/devacc/test-web-project
```

```
INFO: Autodiscovered repositories
       "length": 1,
       "repositories": ["thetradedesk/teams/devacc/test-web-project"]
 WARN: Configured repository is in not in autodiscover list (repository=renovate_extra_flags:)
 INFO: Repository started (repository=thetradedesk/teams/devacc/test-web-project)
       "renovateVersion": "36.24.0"
```

</details>

**Comment by viceice** [comment, marked as answer]
should be in the config.js inside the runner repo

**Reply by greg-slonim** [reply]
@viceice - thank you
